apiclient/src/lib.rs

```diff
@@ -60,43 +64,62 @@ impl std::fmt::Display for DomainOrAddress {
 #[derive(Clone)]
 pub struct ApiClient {
     client: Client,
-    url: String,
+    port: u16,
+    hostname: String,
+    tls_enabled: bool,
 }
 
 impl ApiClient {
     /// Creates a new API client that connects to the given base URL.
     ///
     /// # Arguments
-    /// base_url - The base URL of the server.
-    /// transport_encryption - Whether transport encryption is enabled or not.
+    /// url - The base URL or hostname:port tuple of the server. If the URL
+    /// starts with `https`, TLS will be used. If the URL or hostname:port tuple
+    /// includes a port, that port will be used, otherwise, if TLS is enabled,
+    /// the default port is 443, if TLS is disabled, the default port is 8000.
     ///
     /// # Returns
     /// A new [`ApiClient`].
     pub fn initialize(domain: impl ToString) -> Result<Self, ApiClientInitError> {
+        let mut domain_string = domain.to_string();
+        // If the url doesn't start with http, we assume it's a hostname/port
+        // combination and we prefix it with "http://"
+        if !domain_string.starts_with("http") {
+            domain_string.insert_str(0, "http://");
+        }
```

Comment:
This is precisely the kind of string parsing `Url` should do for us, and we should only look at the URI scheme instead.

I propose to invert the logic here and have HTTPS as a default when no scheme is specified.

Reply:
A valid `Url` always starts with a scheme. However, we want to also accept simple domain names. The purpose of this check is to simple hostnames or (hostname:port) tuples into Urls.

Reply:
As discussed offline:
 - try `Url::parse()` and if it fails with `ParseError::RelativeUrlWithoutBase`, try `Host::parse()` instead. If that succeeds, prepend "https://" to the URL and try `Url::parse()` again.